### PR TITLE
Better makefile

### PR DESCRIPTION
**Pull request by kjdchapman**
Added Guard and changed the Docker configuration to mount the local file system, so that it can watch changes. `make test` now gives you a watching Guard.

`make serve` is now also interactive rather than detached, which is mostly personal preference.

Also includes some general Makefile love (dependencies expressed more tersely, not `serve`-ing when it's not necessary).

### CODE REVIEW

**Comment by kjdchapman** [comment]
Commit messages as written are too terse, @adrianclay ?

**Comment by adrianclay** [comment]
I'd group commits into 1 logical commit, and also make sure each commit has a clear why.  That way when doing `git blame` you can get a bit of understanding for why a line looks like it does.

**Comment by kjdchapman** [comment]
@adrianclay 

Oh, I see. I've tended to put something up for review, then do the rebasing/cleanup after.

If a change is large enough that it's hard to parse in one PR, but still logically belongs together, I'll sometimes do some rebasing to try and make it easier to understand. Generally want to avoid large PRs though.

Is it better for you if I mark these as WIP when they're pre-rebase?

**Comment by kjdchapman** [comment]
In lieu of fixing the Jenkins issues, I'm going to split this up a bit more for different PRs.
